OlinkAnalyze/R/olink_Pathway_Enrichment.R

```diff
@@ -0,0 +1,251 @@
+#' ora_results <- olink_pathway_enrichment(
+#'   data = npx_data1,
+#'   test_results = ttest_results, method = "ORA"
+#' )
+#' }, silent = TRUE)
+#' }
+#' @seealso \itemize{
+#' \item{\code{\link[OlinkAnalyze:olink_pathway_heatmap]{olink_pathway_heatmap}} for generating a heat map of results}
+#' \item{\code{\link[OlinkAnalyze:olink_pathway_visualization]{olink_pathway_visualization}} for generating a bar graph of results}
+#' }
+#'
+#' @importFrom dplyr filter mutate select group_by summarise arrange ungroup distinct pull full_join
+#' @importFrom stringr str_length
+#' @importFrom magrittr %>%
+#' @export
+
+olink_pathway_enrichment <- function(data, test_results, method = "GSEA", ontology = "MSigDb", organism = "human") {
+  # Is Package installed
+  if(!requireNamespace("clusterProfiler", quietly = TRUE) ){
+    stop(" Pathway enrichment requires clusterProfiler package.
+         Please install clusterProfiler before continuing.
+
+         if (!require(\"BiocManager\", quietly = TRUE))
+            install.packages(\"BiocManager\")
+         BiocManager::install(\"clusterProfiler\")")
+  }
+
+  if(!requireNamespace("msigdbr", quietly = TRUE)){
+    stop(" Pathway enrichment requires msigdbr package.
+         Please install msigdbr before continuing.
+
+         install.packages(\"msigdbr\")")
+  }
+  # Data Checks
+
+  if(length(unique(data$OlinkID)) != length(unique(test_results$OlinkID))) {
+    warning("The number of Olink IDs in the data does not equal the number of Olink IDs in the test results.")
+  }
+
+  if("contrast" %in% colnames(test_results) && length(unique(test_results$contrast)) > 1){
+    stop("More than one contrast is specified in test results. Filter test_results for desired contrast.")
+  }
+  if(!("estimate" %in% colnames(test_results))){
+    stop("Estimate column is not present in test results. Please check arguments.")
+  }
+
+  if(!(method %in% c("GSEA", "ORA"))){
+    stop("Method must be \"GSEA\" or \"ORA\".")
+  }
+
+  if(!(ontology %in% c("MSigDb", "Reactome", "KEGG", "GO"))){
+    stop("Ontology must be one of MSigDb, Reactome, KEGG, or GO.")
+  }
+  data2 <- data_prep(data = data)
+  test_results2 <- test_prep(data = data2, test_results = test_results, organism = organism)
+
+  if (method == "ORA") {
+    results <- ora_pathwayenrichment(
+      test_results = test_results2, ontology = ontology,
+      organism = organism)
+    message("Over-representation Analysis performed")
+  } else {
+    geneList <- results_to_genelist(test_results = test_results2)
+    results <- gsea_pathwayenrichment(geneList = geneList, ontology = ontology, organism = organism)
+    message("Gene set enrichment analysis used by default.")
+  }
+  return(results)
+}
+
+
+data_prep <- function(data) {
+  # Filter highest detectibility for repeated IDs
+  olink_ids <- data %>%
+    dplyr::filter(!str_detect(SampleID, "CONTROL*.")) %>%
+    dplyr::filter(QC_Warning == "Pass") %>%
```

Comment:
To cover both T96 and Explore QC_Warning format

```suggestion
    dplyr::filter(toupper(QC_Warning) == "PASS") %>%
```